src/vscode-bicep/src/visualizer/app/components/Graph/style.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { Stylesheet } from "cytoscape";
+
+import moduleSvg from "../../assets/icons/graph/module.svg";
+
+const backgroundColor = "#333333";
+const backgroundColorSecondary = "#3f3f3f";
+const foregroundColor = "#ffffff";
+const foregroundColorSecondary = "#9c9c9c";
+const errorColor = "red";
+
+function escapeXml(text: string) {
+  return text.replace(/[<>&'"]/g, (c) => {
+    switch (c) {
+      case "<":
+        return "&lt;";
+      case ">":
+        return "&gt;";
+      case "&":
+        return "&amp;";
+      case "'":
+        return "&apos;";
+      case '"':
+        return "&quot;";
+      default:
+        return c;
+    }
+  });
+}
+
+function truncate(text: string, lengthLimit: number) {
+  if (text.length <= lengthLimit) {
+    return text;
+  }
+
+  const charsLength = lengthLimit - 3;
+  const headLength = Math.ceil(charsLength / 2);
+  const tailLength = Math.floor(charsLength / 2);
+
+  return (
+    text.substr(0, headLength) + "..." + text.substr(text.length - tailLength)
+  );
+}
+
+function createDataUri(svg: string) {
+  const domParser = new DOMParser();
+  const svgElement = domParser.parseFromString(svg, "text/xml").documentElement;
+
+  return "data:image/svg+xml;utf8," + encodeURIComponent(svgElement.outerHTML);
+}
+
+export async function createChildlessNodeBackgroundUri(
+  symbol: string,
+  type: string,
+  isCollection: boolean
+): Promise<string> {
+  const iconSvg =
+    type !== "<module>"
+      ? (await import("../../assets/icons/graph/resource.svg")).default
+      : moduleSvg;
+
+  type = type.split("/").pop() ?? type;
+  type += isCollection ? "[]" : "";
+
+  const backgroundSvg = `<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE svg>
```

Comment:
Why not just use tsx here? Wouldn't that remove the need for manual xml escaping?

Reply:
Because we are using a vanilla JS library ([cytoscape.js](https://js.cytoscape.org/)) to render the graph and this is the way to set custom SVG background for nodes.

Reply:
PS: we are using cytoscape because it is capable of rendering compound/nested graph. Eventually I would like to drop the dependency on cytoscape and implement our own graph renderer as a React component, but that's something we can do in the future as it requires quite a lot of work (or, we use an existing React library like [react-flow](https://reactflow.dev/), once they support rendering compound/nested graph).